tasks/indexing/task_ipni.go

```diff
@@ -0,0 +1,452 @@
+package indexing
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log/v2"
+	carv2 "github.com/ipld/go-car/v2"
+	"github.com/ipld/go-car/v2/index"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipni/go-libipni/ingest/schema"
+	"github.com/ipni/go-libipni/metadata"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/curio/deps/config"
+	"github.com/filecoin-project/curio/harmony/harmonydb"
+	"github.com/filecoin-project/curio/harmony/harmonytask"
+	"github.com/filecoin-project/curio/harmony/resources"
+	"github.com/filecoin-project/curio/lib/ffi"
+	"github.com/filecoin-project/curio/lib/indexstore"
+	"github.com/filecoin-project/curio/lib/ipni/chunker"
+	"github.com/filecoin-project/curio/lib/ipni/ipniculib"
+	"github.com/filecoin-project/curio/lib/passcall"
+	"github.com/filecoin-project/curio/lib/pieceprovider"
+
+	"github.com/filecoin-project/lotus/storage/sealer/storiface"
+)
+
+var ilog = logging.Logger("ipni")
+
+type IPNITask struct {
+	db            *harmonydb.DB
+	indexStore    *indexstore.IndexStore
+	pieceProvider *pieceprovider.PieceProvider
+	sc            *ffi.SealCalls
+	cfg           *config.CurioConfig
+}
+
+func NewIPNITask(db *harmonydb.DB, sc *ffi.SealCalls, indexStore *indexstore.IndexStore, pieceProvider *pieceprovider.PieceProvider, cfg *config.CurioConfig) *IPNITask {
+
+	return &IPNITask{
+		db:            db,
+		indexStore:    indexStore,
+		pieceProvider: pieceProvider,
+		sc:            sc,
+		cfg:           cfg,
+	}
+}
+
+func (I *IPNITask) Do(taskID harmonytask.TaskID, stillOwned func() bool) (done bool, err error) {
+	ctx := context.Background()
+
+	var tasks []struct {
+		SPID   int64                   `db:"sp_id"`
+		Sector abi.SectorNumber        `db:"sector"`
+		Proof  abi.RegisteredSealProof `db:"reg_seal_proof"`
+		Offset int64                   `db:"sector_offset"`
+		CTXID  []byte                  `db:"context_id"`
+		RM     bool                    `db:"is_rm"`
```

Comment:
Naming suggestions since these are not abbreviations:
```suggestion
		CtxID  []byte                  `db:"context_id"`
		Rm     bool                    `db:"is_rm"`
```